artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java

```diff
@@ -3381,10 +3382,23 @@ synchronized void initialisePart2(boolean scalingDown) throws Exception {
 
       pagingManager.reloadStores();
 
-      JournalLoadInformation[] journalInfo = loadJournals();
+      Set<Long> storedLargeMessages = new HashSet<>();
+      JournalLoadInformation[] journalInfo = loadJournals(storedLargeMessages);
 
       if (rebuildCounters) {
-         pagingManager.rebuildCounters();
+         pagingManager.rebuildCounters(storedLargeMessages);
+
+         pagingManager.execute(() -> {
+            storedLargeMessages.forEach(id -> {
+               try {
+                  SequentialFile file = storageManager.createFileForLargeMessage(id, true);
+                  logger.debug("Removing pending large message for file={}", file);
+                  file.delete();
+               } catch (Exception e) {
+                  logger.warn(e.getMessage(), e);
```

Comment:
Might be good to give some context in the log message about what was being done when the exception occurred. May not be so obvious to a user from the exception message itself.